XLS-0065d-single-asset-vault/README.md

```diff
@@ -0,0 +1,653 @@
+
+The "paper loss" temporarily decreases the vault value. A malicious depositor may take advantage of this to deposit assets at a lowered price and withdraw them once the price increases.
+
+Consider a vault with a total value of $1.0m and total shares of $1.0m. Assume the "paper loss" for the vault is $900k. The new exchange rate is as follows:
+
+$$
+exchangeRate = \frac{AssetsTotal - LossUnrealized}{SharesTotal}
+$$
+
+$$
+exchangeRate = \frac{1,000,000 - 900,000}{1,000,000} = 0.1
+$$
+
+After the "paper loss" is cleared, the new effective exchange rate would be as follows:
+
+$$
+exchangeRate = \frac{AssetsTotal}{SharesTotal}
+$$
+
+$$
+exchangeRate = \frac{1,000,000}{1,000,000} = 1.0
+$$
+
+A depositor could deposit $100k assets at a 0.1 exchange rate and get 1.0m shares. Once the "paper loss" is cleared, their shares would be worth $1.0m.
+
+To account for this problem, the Vault must use two different exchange rate models: one for depositing assets and one for withdrawing them.
+
+##### 2.1.7.2 Exchange Rate Algorithms
+
+**Variables**
+
+The following variables define the Vault balance:
+
+- $\Gamma_{share}$ - the total number of shares issued by the vault.
+- $\Gamma_{asset}$ - the total assets in the vault, including any future yield.
+
+- $\Delta_{asset}$ - the change in the total amount of assets after a deposit, withdrawal, or redemption.
+- $\Delta_{share}$ - che change in the total amount of shares after a deposit, withdrawal, or redemption.
+
+- $\iota$ - the unrealized loss of the vault.
+
+###### 2.1.7.2.1 **Deposit**
+
+We compute the number of shares ($\Delta_{share}$) a depositor will receive as follows:
+
+$$\Delta_{share} = \frac{\Delta_{asset} \times (\Gamma_{share} + 0^{\Gamma_{share}})}{\Gamma_{asset} + 0^{\Gamma_{asset}}} $$
+
+The following equations govern the updated vault composition after a successful deposit:
+
+- $\Gamma_{asset} = \Gamma_{asset} + \Delta_{asset}$ New balance of assets in the vault.
+- $\Gamma_{share} = \Gamma_{share} + \Delta_{share}$ New share balance in the vault.
+
+###### 2.1.7.2.2 **Redeem**
+
+We compute the number of assets ($\Delta_{asset}$) returned by burning $\Delta_{share}$ as follows:
+
+$$\Delta_{asset} = \frac{\Delta_{share} \times (\Gamma_{asset} - \iota + 0^{asset})}{\Gamma_{share} + 0^{\Gamma_{share}}} $$
+
+The following equations govern the updated vault composition after a successful redemption:
+
+- $\Gamma_{asset} = \Gamma_{asset} - \Delta_{asset}$ New balance of assets in the vault.
+- $\Gamma_{share} = \Gamma_{share} - \Delta_{share}$ New share balance in the vault.
+
+###### 2.1.7.2.3 **Withdraw**
+
+We compute the number of shares to burn to withdraw $\Delta_{asset}$ as follows:
+
+$$\Delta_{share} = \frac{\Delta_{asset} \times (\Gamma_{share} + 0^{\Gamma_{share}})}{\Gamma_{asset} + \iota + 0^{\Gamma_{asset}}} $$
```

Comment:
```suggestion
$$\Delta_{share} = \frac{\Delta_{asset} \times (\Gamma_{share} + 0^{\Gamma_{share}})}{\Gamma_{asset} - \iota + 0^{\Gamma_{asset}}} $$
```

Reply:
Great catch! I discounted, not increased the value! :facepalm: